### DISCUSSION TITLE
How to get a resource by name

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#8421] [Q&A] [answered]
*Getting a resource's dependent resource by name*

I'm using a factory function to create a resource, something like
```
def storage_bucket(dependent_resource_name):
    @resource(
        required_resource_keys={dependent_resource_name},
    )
    def _storage_bucket(init_context):
        credentials = init_context.resources[dependent_resource_name]
        return StorageBucket(credentials)
    return _storage_bucket
```
The problem is in the line `credentials = init_context.resources[dependent_resource_name]` which is giving me the error `TypeError: tuple indices must be integers or slices, not str`. I'm aware that `init_context.resources`  is a `_ScopeResources`. I also see that `_ScopedResource` implements `__getitem__`, which is why I assumed this would work.
I know that I could call `_ScopedResources._asdict()` and the get the resource by name.
Can anyone suggest the :dagster: way to do this?

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C01U954MEER/p1643861142327069?thread_ts=1643861142.327069&cid=C01U954MEER).*

**Comment by yuhan** [comment, marked as answer]
You can use `getattr`, for example:

```python
from dagster import op, job, ResourceDefinition


@op(required_resource_keys={"a", "b", "c"})
def my_op(context):
    print(getattr(context.resources, "a"))  # 1
    print(getattr(context.resources, "b"))  # 2
    print(getattr(context.resources, "c"))  # 3


@job(
    resource_defs={
        "a": ResourceDefinition.hardcoded_resource(1),
        "b": ResourceDefinition.hardcoded_resource(2),
        "c": ResourceDefinition.hardcoded_resource(3),
    }
)
def my_job():
    my_op()

```
